src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+    let sepp = separator;
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    if (separator == null) {
+      sepp = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
```

Comment:
Make one check for null and undefined in the beginning of your loop. Don't repeat yourself.